nixui/options/option_definition.py

```diff
@@ -118,3 +190,66 @@ def get_expression(obj):
         return "null"
     else:
         raise TypeError(str((type(obj), obj)))
+
+
+#############################
+# Expression string to object
+#############################
+def expression_node_to_python_object(value_node, context):
+    if value_node.name == 'NODE_LIST':
+        # recursively generate list object
+        return [
+            expression_node_to_python_object(child_node, context)
+            for child_node in value_node.elems
+            if isinstance(child_node, syntax_tree.Node)
+        ]
+
+    elif value_node.name == 'NODE_STRING':
+        assert value_node.elems[0].name == 'TOKEN_STRING_START'
+        assert value_node.elems[-1].name == 'TOKEN_STRING_END'
+        python_obj = ''
+        for child_node in value_node.elems[1:-1]:
+            if child_node.name == 'TOKEN_STRING_CONTENT':
+                python_obj += child_node.quoted
+            elif child_node.name == 'NODE_STRING_INTERPOL':
+                return Unresolvable  # TODO: handle string interpolation
+            else:
+                return Unresolvable  # TODO
+        return python_obj
+
+    elif value_node.name == 'NODE_IDENT':
+        if len(value_node.elems) == 1 and value_node.elems[0].name == 'TOKEN_IDENT':
+            if value_node.elems[0].quoted == 'true':
+                return True
+            elif value_node.elems[0].quoted == 'false':
+                return False
+            else:
+                return Unresolvable  # TODO
+
+    elif value_node.name == 'NODE_LITERAL':
+        if len(value_node.elems) == 1:
+            if value_node.elems[0].name == 'NODE_PATH':
+                return Unresolvable  # TODO
+            if value_node.elems[0].name == 'TOKEN_PATH':
+                return Path(
+                    value_node.elems[0].quoted,
+                    context['module_dir'],
+                )
+            elif value_node.elems[0].name == 'TOKEN_INTEGER':
+                return int(value_node.elems[0].quoted)
+            elif value_node.elems[0].name == 'TOKEN_URI':
+                return str(value_node.elems[0].quoted)
+            elif value_node.elems[0].name == 'TOKEN_FLOAT':
+                return float(value_node.elems[0].quoted)
```

Comment:
else raise `ValueError`

Reply:
disregard, all unmatched statements result in `ValueError` per the last line of the fn